### PR TITLE
Spatial index cleanup

### PR DESCRIPTION
**Pull request by OliviaYtterbrink**
Changed to do sampling at SpatialFusion level and got rid of the need to have an init method in `SpatialCRSSchemaIndex`.

### CODE REVIEW

File: community/kernel/src/test/java/org/neo4j/kernel/impl/index/schema/fusion/SpatialFusionIndexPopulatorTest.java [hunk 1]
```diff
@@ -196,47 +198,48 @@ public void markAsFailedMustMarkAll() throws Exception
...
-            fusionIndexPopulator.includeSample( update );
-
-            // then
-            SpatialKnownIndex spatialKnownIndex = indexMap.get( point.getCoordinateReferenceSystem() );
-            verify( spatialKnownIndex ).includeSample( update );
-            reset( spatialKnownIndex );
-        }
-    }
+//    @Test
+//    public void shouldIncludeSampleOnCorrectPopulator() throws Exception
```
Comment (burqen) [inline community/kernel/src/test/java/org/neo4j/kernel/impl/index/schema/fusion/SpatialFusionIndexPopulatorTest.java]: Oops 🙈

File: enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/SpatialFunctionsAcceptanceTest.scala [hunk 2]
```diff
@@ -19,18 +19,15 @@
...
 import org.neo4j.graphdb.spatial.Point
 import org.neo4j.internal.cypher.acceptance.CypherComparisonSupport._
 import org.neo4j.values.storable.{CoordinateReferenceSystem, Values}
 
 class SpatialFunctionsAcceptanceTest extends ExecutionEngineFunSuite with CypherComparisonSupport {
 
-  val pointConfig = Configs.Interpreted - Configs.Version2_3
-  val equalityConfig = Configs.Interpreted - Configs.OldAndRule
+  val pointConfig: TestConfiguration = Configs.Interpreted - Configs.Version2_3
+  val equalityConfig: TestConfiguration = Configs.Interpreted - Configs.OldAndRule
```
Comment (Lojjs) [inline enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/SpatialFunctionsAcceptanceTest.scala]: Can you remove type annotation here?

File: community/kernel/src/main/java/org/neo4j/kernel/impl/api/index/sampling/IndexSamplingConfig.java [hunk 3]
```diff
@@ -30,9 +30,17 @@
...
-        this.sampleSizeLimit = config.get( GraphDatabaseSettings.index_sample_size_limit );
-        this.updateRatio = ((double) config.get( GraphDatabaseSettings.index_sampling_update_percentage )) / 100.0d;
-        this.backgroundSampling = config.get( GraphDatabaseSettings.index_background_sampling_enabled );
+        this( config.get( GraphDatabaseSettings.index_sample_size_limit ),
+              config.get( GraphDatabaseSettings.index_sampling_update_percentage ) / 100.0d,
+              config.get( GraphDatabaseSettings.index_background_sampling_enabled )
+        );
+    }
+
+    public IndexSamplingConfig( int sampleSizeLimit, double updateRatio, boolean backgroundSampling )
```
Comment (Lojjs) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/api/index/sampling/IndexSamplingConfig.java]: This new constructor seems only to be used from the one above. What was the reason for splitting it up?

**Review by Lojjs — COMMENTED** [review]
Lot of good clean-up here! I especially liked that you have got rid of the `NONE` state and make init simpler. Just one super-small comment and one question:

File: community/kernel/src/main/java/org/neo4j/kernel/impl/api/index/sampling/IndexSamplingConfig.java (hunk 3 above)
Comment (OliviaYtterbrink) [reply community/kernel/src/main/java/org/neo4j/kernel/impl/api/index/sampling/IndexSamplingConfig.java]: I think it was added to help make some testing easier, but it seems we reverted that part without reverting this. I'll remove it while rebasing.

**Review by Lojjs — APPROVED** [review]

**Pull request merged**